src/Testcontainers/Containers/DockerContainer.cs

```diff
@@ -507,8 +508,15 @@ protected virtual async Task UnsafeStopAsync(CancellationToken ct = default)
       await _client.StopAsync(_container.ID, ct)
         .ConfigureAwait(false);
 
-      _container = await _client.Container.ByIdAsync(_container.ID, ct)
-        .ConfigureAwait(false);
+      try
+      {
+        _container = await _client.Container.ByIdAsync(_container.ID, ct)
+          .ConfigureAwait(false);
+      }
+      catch (DockerApiException)
+      {
+        _container = null;
+      }
```

Comment:
👍 I think this should be:

```suggestion
      catch (DockerApiException)
      {
        _container = new ContainerInspectResponse();
      }
```

We can apply the same changes to the image, network and volume operation classes. I can do it in a follow-up PR.

Reply:
I think setting to `null` is correct (it was the previous behavior) because the `Exists` method just below checks `_container != null`.

> We can apply the same changes to the image, network and volume operation classes. I can do it in a follow-up PR.

I tried this but it needs some more adaptation too: https://github.com/testcontainers/testcontainers-dotnet/runs/29692350906

Reply:
I remembered it was like that before: https://github.com/testcontainers/testcontainers-dotnet/commit/6e31cf387c279df3be6e47615a01c3076011c8fc. IIRC it was/is not supposed to be null. We set it to an instance in delete similar image, network and volume. I can double-check it later.

Reply:
> I tried this but it needs some more adaptation too: https://github.com/testcontainers/testcontainers-dotnet/runs/29692350906

This is based on how we determine if an image already exists and whether we need to pull it. I updated the PR, LMKWYT. I will merge it if you are good with the changes